File: 노영주/37주차/단속카메라.js
function solution(routes) {
  routes.sort((a, b) => a[0] - b[0] || a[1] - b[1]);

  let answer = 1;
  let coordinate = routes[0][1];

  for (let index = 1; index < routes.length; index += 1) {
    const [entry, exit] = routes[index];
    if (entry > coordinate) {
      answer += 1;
      coordinate = exit;
    } else {
      coordinate = Math.min(coordinate, exit);
    }
  }

  return answer;
}

File: 윤호선/37주차/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 1;
    sort(routes.begin(),routes.end());
    int y=routes[0][1];
    
    for(int i=1;i<routes.size();i++){
        int start=routes[i][0];
        int end=routes[i][1];
        
        if(y<start){
            ++answer;
            y=end;
        }
        else{
            if(end<y)y=end;
        }
    }
    
    return answer;
}